From d36c7994fed7667a3a0b17be1e62fe235213e979 Mon Sep 17 00:00:00 2001
Message-Id: <d36c7994fed7667a3a0b17be1e62fe235213e979.1435267675.git.erack@redhat.com>
From: Eike Rathke <erack@redhat.com>
Date: Thu, 25 Jun 2015 22:54:37 +0200
Subject: [PATCH] tdf#44419 in second reference part stop number parsing at
 separator
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------erAck-patch-parts"

This is a multi-part message in MIME format.
--------------erAck-patch-parts
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit


Change-Id: I70218bc41df0d56ab42d652aa7ac51733dc06f4b
(cherry picked from commit 71dba79c1c84d693a59c53a8965caef0ddd1c2cc)
---
 sc/source/core/tool/compiler.cxx | 12 ++++++++++++
 1 file changed, 12 insertions(+)


--------------erAck-patch-parts
Content-Type: text/x-patch; name="0001-tdf-44419-in-second-reference-part-stop-number-parsi.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0001-tdf-44419-in-second-reference-part-stop-number-parsi.patch"

diff --git a/sc/source/core/tool/compiler.cxx b/sc/source/core/tool/compiler.cxx
index 0e81896..1f2e4de 100644
--- a/sc/source/core/tool/compiler.cxx
+++ b/sc/source/core/tool/compiler.cxx
@@ -2372,6 +2372,18 @@ Label_MaskStateMachine:
             }
             else
             {
+                // When having parsed a second reference part, ensure that the
+                // i18n parser did not mistakingly parse a number that included
+                // a separator which happened to be meant as a parameter
+                // separator instead.
+                if (mnRangeOpPosInSymbol >= 0 && (aRes.TokenType & KParseType::ASC_NUMBER))
+                {
+                    for (sal_Int32 i = nSrcPos; i < aRes.EndPos; ++i)
+                    {
+                        if (pStart[i] == cSep)
+                            aRes.EndPos = i;    // also ends for
+                    }
+                }
                 aSymbol.append( pStart + nSrcPos, aRes.EndPos - nSrcPos);
                 nSrcPos = aRes.EndPos;
                 c = pStart[nSrcPos];

--------------erAck-patch-parts--


